gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -1074,15 +1096,15 @@ boolean notRequiredByAnyTx( LocalRegion region, RegionEntry re,
       Set<TXId> othersTxIds = new UnifiedSet<TXId>(4);
       for (TXStateProxy txProxy : getTxManager().getHostedTransactionsInProgress()) {
         TXState txState = txProxy.getLocalTXState();
-        if ((txState != null && !txState.isClosed() && TXState.checkEntryInSnapshot
+        if ((txState != null && !txState.isClosed() && TXState.checkEntryInSnapshotWithoutMyChange
```

Comment:
What's "checkEntryInSnapshotWithoutMyChange"? Please change the name to reflect what it is supposed to be.

Reply:
The check doesn't consider changes done by tx as in snapshot. This is to avoid a case where non-committed entry can cause committed entry to be removed.